operator/pkg/reconciliation/reconcile_racks.go

```diff
@@ -1609,6 +1682,85 @@ func (rc *ReconciliationContext) CheckRollingRestart() result.ReconcileResult {
 	return result.Continue()
 }
 
+func (rc *ReconciliationContext) setCondition(condition *api.DatacenterCondition) bool {
+	dc := rc.Datacenter
+	if dc.GetConditionStatus(condition.Type) != condition.Status {
+		// We are changing the status, so record the transition time
+		condition.LastTransitionTime = metav1.Now()
+		dc.SetCondition(*condition)
+		return true
+	}
+	return false
+}
+
+func (rc *ReconciliationContext) CheckConditionInitializedAndReady() result.ReconcileResult {
+	dc := rc.Datacenter
+	dcPatch := client.MergeFrom(dc.DeepCopy())
+	logger := rc.ReqLogger
+	
+	updated := false
+	updated = rc.setCondition(
+		api.NewDatacenterCondition(api.DatacenterInitialized, corev1.ConditionTrue)) || updated
+	
+	if dc.GetConditionStatus(api.DatacenterStopped) == corev1.ConditionFalse {
+		updated = rc.setCondition(
+			api.NewDatacenterCondition(api.DatacenterReady, corev1.ConditionTrue)) || updated
+	}
+
+	if updated {
+		err := rc.Client.Status().Patch(rc.Ctx, dc, dcPatch)
+		if err != nil {
+			logger.Error(err, "error patching datacenter status")
+			return result.Error(err)
+		}
+
+		// We may have ignored some changes before becoming ready. Ensure the reconcile loop
+		// gets a chance to run again to pick up anything missed.
+		return result.RequeueSoon(0)
+	}
+
+	return result.Continue()
+}
+
+func (rc *ReconciliationContext) CheckClearActionConditions() result.ReconcileResult {
+	dc := rc.Datacenter
+	logger := rc.ReqLogger
+	dcPatch := client.MergeFrom(dc.DeepCopy())
+
+	// If we are here, any action that was in progress should now be completed, so start
+	// clearing conditions
+	actionConditionTypes := []api.DatacenterConditionType{
+		api.DatacenterReplacingNodes,
+		api.DatacenterScalingUp,
+		api.DatacenterUpdating,
+		api.DatacenterRollingRestart,
+		api.DatacenterResuming,
+	}
+	updated := false
+	for _, conditionType := range(actionConditionTypes) {
+		updated = rc.setCondition(
+			api.NewDatacenterCondition(conditionType, corev1.ConditionFalse)) || updated
+	}
+
+	if updated {
+		err := rc.Client.Status().Patch(rc.Ctx, dc, dcPatch)
+		if err != nil {
+			logger.Error(err, "error patching datacenter status")
+			return result.Error(err)
+		}
+
+		// There may have been changes to the CassandraDatacenter resource that we ignored
+		// while executing some action on the cluster. For example, a user may have 
+		// requested to scale up the node count while we were in the middle of a rolling
+		// restart. To account for this, we requeue to ensure reconcile gets called again
+		// to pick up any such changes that we ignored previously.
+		return result.RequeueSoon(0)
```

Comment:
are we already "ignoring" stuff?

Reply:
Err not yet. Once we start doing things like postponing a scale up until a rolling restart is done, that line will become important. However, I ended up not doing the other half of this where the operator consumes the conditions to inform what it does and when.

Would you prefer I remove it?

Reply:
nah, leave it there, I just wanted to know